heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/MetricsManager.java

```diff
@@ -332,87 +332,96 @@ public void start() {
      * Handler for uncaughtException
      */
     public void uncaughtException(Thread thread, Throwable exception) {
-      // We would fail fast when errors occur
-      if (exception instanceof Error) {
-        LOG.log(Level.SEVERE,
-            "Error caught in thread: " + thread.getName()
-                + " with thread id: " + thread.getId() + ". Process halting...",
-            exception);
-        Runtime.getRuntime().halt(1);
-      }
+      // Add try and catch block to prevent new exceptions stop the handling thread
+      try {
```

Comment:
Generally try blocks should be small and this one is pretty large. Could you move the logic inside the block into a private method and just delegate to that inside the try?